llm/chatbot.py

```
@@ -0,0 +1,36 @@
+for message in st.session_state.messages:
+    with st.chat_message(message["role"]):
+        st.write(message["content"])
+
+# User Input and Response Generation
+if prompt := st.chat_input():
+    st.session_state.messages.append({"role": "user", "content": prompt})
+    with st.chat_message("user"):
+        st.write(prompt)
+
+    with st.chat_message("assistant"):
+        with st.spinner("Thinking..."):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)